src/Modules/Meetings/Domain/MeetingComments/MeetingCommentService.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Threading.Tasks;
+using CompanyName.MyMeetings.BuildingBlocks.Domain;
+using CompanyName.MyMeetings.Modules.Meetings.Domain.MeetingComments.Rules;
+using CompanyName.MyMeetings.Modules.Meetings.Domain.MeetingGroups;
+using CompanyName.MyMeetings.Modules.Meetings.Domain.MeetingMemberCommentLikes;
+using CompanyName.MyMeetings.Modules.Meetings.Domain.Members;
+
+namespace CompanyName.MyMeetings.Modules.Meetings.Domain.MeetingComments
+{
+    public class MeetingCommentService
```

Comment:
This should be in the Application layer, because it is an impure service and it does not represent any business concept. 

Domain Service in Domain should - take some inputs and returns the output - everything based on Domain objects. 

If you want to implement this logic in Domain, you need to set `meetingMemberCommentLikesCount` in Application and pass it as an argument to the method of Domain Object. This way you don't need to mock the repository.

Of course sometimes it is better to keep this logic in application - mostly for performance reasons.

Reply:
Thanks for your opinion! I get it and will change implementation. As for current implementation I would like do explain my decision:
1. I wanted make Domain model incapsulated(completed). Now it provides to application layer one way to like comment(call `MeetingCommentService` method) and keeps all invariants.
1. I get known about such approach(with impure Domain service) from this article of Vladimir Khorikov. Although he describes the approach he nevertheless tends to keep Domain model free of out-of-process dependencies and make model incomplete but pure. But I decided that inpure Domain service is not so bad if its only dependency is `IMeetingMemberCommentLikesRepository` and as far as I know an interface of repository is considered as part of Domain model.

What do you think about these points? 

And returning to your comment, what do you prefer: place logic in application layer or pass `meetingMemberCommentLikesCount ` as argument?  I think it's better to check the rule in application layer because `meetingMemberCommentLikesCount` as a `AddLike` method's parameter will make `MeetingComment`'s abstraction leaky imo.

Reply:
Hi Andrei,

In actual implementation `MeetingCommentService` does not have any business logic in my opinion (only orchestrate application workflow as handlers). I agree with Vladimir that when there is some logic sometimes it is good to extract it to separate service but I would say this is not the case here in my opinion :)

About solution - I always try to put as much as possible logic in the Domain so I would pass it raw data as an argument to the `Like` method. 

Of course, sometimes it is not practical because of performance - then you can pass some abstraction like `ICommentValidator` or make this validation in the application layer.

Reply:
Ok, thanks for explanation!